### PR TITLE
[PY-627] Long Video support for split video annotations

### PR DESCRIPTION
**Pull request by Nathanjp91**
# Problem
Long videos currently does not get split correctly for formats like coco that require treating each image seperately. 

# Solution
Added in logic for generation of split video annotations if frame_urls is not present, relying instead on frame_count, which is a field to be added in conjunction with changes to slots

# Changelog
split video annotations should now work correctly with long video files

### CODE REVIEW

File: darwin/utils/utils.py [hunk 1]
```diff
@@ -993,13 +994,17 @@ def split_video_annotation(annotation: dt.AnnotationFile) -> List[dt.AnnotationF
...
-        raise AttributeError("This Annotation has no frame urls")
-
+        raise AttributeError("This is not a video annotation")
+
+    if not annotation.frame_count and not annotation.frame_urls:
+        raise AttributeError("This Annotation has no frames")
+    urls = (
+        annotation.frame_urls
+        if annotation.frame_urls
+        else [None] * (annotation.frame_count or 1)
```
Comment (Nathanjp91) [inline darwin/utils/utils.py:1004]: the `or 1` here is just to satisfy type check, the prior assert should already catch that either frame_count or frame_urls should exist but for some reason mypy is not smart enough to pick that up.

File: darwin/utils/utils.py [hunk 2]
```diff
@@ -993,13 +994,17 @@ def split_video_annotation(annotation: dt.AnnotationFile) -> List[dt.AnnotationF
...
+    )
     frame_annotations = []
-    for i, frame_url in enumerate(annotation.frame_urls):
+    for i, frame_url in enumerate(urls):
```
Comment (Nathanjp91) [inline darwin/utils/utils.py:1007]: In the case of a long video frame_url get's passed into the AnnotationFile.image_url field as None. Perhaps this is not the correct assumption but I couldn't reason about what an image_url would be in this case, and none of our tests fail.

File: darwin/utils/utils.py (hunk 1 above)
Comment (rslota) [reply darwin/utils/utils.py:1004]: Can we add a comment somewhere here on why we're chaining this? Long Videos are pretty new, so this code may be a bit confusing without context.

Comment (saurbhc) [inline darwin/utils/utils.py:1004]: Just a bit pythonic this way with ternary operator:
```suggestion
    urls = annotation.frame_urls or [None] * (annotation.frame_count or 1)
```

**Review by saurbhc — APPROVED** [review]

**Review by rslota — APPROVED** [review]

**Pull request merged**